### DISCUSSION TITLE
NodeJS changes to handling of headers

### DISCUSSION THREAD

**Discussion by hardillb** [sindresorhus/got#2080] [Ideas]
Given https://github.com/nodejs/node/issues/43798, any chance of reviewing this comment:

> Got doesn't support insecureHTTPParser and never will. signal is a to-do.

https://github.com/sindresorhus/got/issues/1794#issuecomment-886050367

While yes, every HTTP server should follow the spec, there are a huge number of broken devices and code out there, having the option to be a little bit more lenient on client side makes working with those things that in most cases can not ever (e.g. discontinued obsolete web cams) be fixed possible. This helps save on the ever growing mountain of electronic waste.

p.s. adding it as a back port to the 11.x line would be amazing as well.

**Comment by szmarczak** [comment]
This is rather unlikely. You can use `node-fetch` instead which supports this: https://github.com/node-fetch/node-fetch#insecure-http-parser

**Reply by hardillb** [reply]
Changing the http client is not an option until the next major release (next year)

**Reply by szmarczak** [reply]
Alternatively in `beforeRequest` hook you can do something like this: https://github.com/apify/got-scraping/blob/master/src/hooks/insecure-parser.ts
